SingularityService/src/main/java/com/hubspot/singularity/smtp/MailTemplateHelpers.java

```diff
@@ -142,21 +158,42 @@ public MailTemplateHelpers(SandboxManager sandboxManager, SingularityConfigurati
     }
   }
 
-  private Optional<Long> getMaybeOffset(String slaveHostname, String fullPath, String filename, Optional<String> directory) {
+  // Searches through the file,
+  private Optional<Long> getMaybeOffset(final String slaveHostname, final String fullPath, final Long logLength, Optional<SingularityTask> task) {
+    long offset = 0;
+    String regex = getLogErrorRegex(task);
+    long length = logLength + regex.length(); // Get extra so that we can be sure to find the error
+    Pattern pattern = Pattern.compile(regex);
```

Comment:
What happens if an invalid pattern is supplied? We should probably catch this and default to returning the tail of the log file (like old behavior)